compiler/rustc_target/src/target_features.rs

```diff
@@ -435,4 +437,11 @@ impl super::spec::Target {
             _ => &[],
         }
     }
+
+    pub fn dependant_target_features(&self) -> &'static [(&'static str, &'static str)] {
+        match &*self.arch {
+            "wasm32" | "wasm64" => WASM_DEPENDANT_FEATURES,
+            _ => &[],
+        }
+    }
```

Comment:
It might be worth adding some extra documentation to this method explaining the meaning of the returned array as well along with what each value in the pair means

Reply:
https://github.com/rust-lang/rust/blob/47a4845dedd6d05a89eb4c46db90091006bb8455/compiler/rustc_target/src/target_features.rs#L441
Happy to take any improvements on that English.
I could also make a dedicated Struct instead of that tuple?

Reply:
I think the wording you added is reasonable. I don't have a personal preference on struct-or-not.